physics/discrete_trajectory.hpp

```diff
@@ -50,6 +50,9 @@ class DiscreteTrajectoryIterator
     DegreesOfFreedom<Frame> const& degrees_of_freedom;
   };
 
+  static reference MakeReference(
+      typename DiscreteTrajectoryTraits<Frame>::TimelineConstIterator it);
```

Comment:
This factory is a bit tasteless.  I would say that the initializer lists `{it->first, it->second}` that we had previously was perfectly readable (if slightly verbose) and did not clutter the API like this function does.

Reply:
`{it->first, it->second}` does not work for `FakeTrajectoryIterator::reference` (a.k.a. `Instant const&`) in `forkable_test.cc`. How do you recommend I proceed?

Reply:
Ah, that's annoying.  Do you understand why Visual Studio was complaining about the constructor?  That seemed like a better approach than the factory.

If the constructor that takes a `TimelineConstIterator` doesn't work, would a constructor that takes a `Timeline::value_type` (i.e., the pair itself) by reference work?

Reply:
I suspect it is because I forgot about `FakeTrajectory` and didn't handle it at all until 89ca90a.

I'm trying the constructor approach again but this time replacing `FakeTrajectoryIterator::reference` with a wrapper struct with the proper constructor. Hopefully Visual Studio is happy this time.